plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -284,7 +301,7 @@ object StmtConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext>()
         return when (typeOperatorCall.operation) {
             FirOperation.IS -> Is(argument, conversionType)
             FirOperation.NOT_IS -> Not(Is(argument, conversionType))
-            FirOperation.AS -> Cast(argument, conversionType).withAccessAndProvenInvariants()
+            FirOperation.AS -> argument.withNewTypeAccessAndProvenInvariants(conversionType)
```

Comment:
Is this a fix of an earlier bug?
I understand it can be a bit of a pain, but I would encourage splitting the small unrelated changes into a separate PR so that that can be accepted quickly.

Reply:
OK, I think it was just some optimization that would rarely work anyway.
Okay, I'll trying doing so.